x-pack/plugins/gis/public/shared/layers/layer.js

```diff
@@ -33,6 +32,7 @@ export class AbstractLayer {
   static createDescriptor(options = {}) {
     const layerDescriptor = { ...options };
 
+    layerDescriptor.source = this._source;
```

Comment:
Why are you setting `descriptor.source` to an object instance? I thought the descriptors just held pure data and not instances. I don't think this is needed and may cause problems when storing the layer in the store since `this._source` is not serializable.